devolutions-gateway/src/credendials.rs

```diff
@@ -0,0 +1,82 @@
+use core::fmt;
+
+use serde::{de, ser};
+
+#[derive(Debug, Deserialize)]
+#[serde(tag = "kind")]
+pub(crate) enum Credentials {
+    #[serde(rename = "username-password")]
+    UsernamePassword { username: String, password: Password },
+}
```

Comment:
@awakecoding Here, we are assuming TLS is used, otherwise the credentials are pushed in plain text. Do you want an extra layer of encryption? (We’ll could use the delegation private key for that.)